pipelines/postprocess_scoring.R

```diff
@@ -39,139 +39,121 @@ purrr::walk(script_packages, \(pkg) {
+  summarised_rel <- filtered_scores |>
     get_pairwise_comparisons(
-      baseline = "cdc_baseline",
-      by = "location"
+      baseline = baseline,
+      by = by
     ) |>
-    group_by(model, location) |>
-    summarise(rel_wis = mean(wis_scaled_relative_skill))
+    filter(.data$compare_against == !!baseline) |>
+    select(model,
+      all_of(by),
+      relative_wis =
+        "wis_scaled_relative_skill"
+    )
 
-  abs_wis <- quantile_scores |>
-    filter(scale == !!scale) |>
-    summarise_scores(by = c("model", "location")) |>
+  summarised <- filtered_scores |>
+    summarise_scores(by = c("model", by)) |>
     select(model,
-      location,
+      all_of(by),
       abs_wis = wis,
       mae = ae_median,
       interval_coverage_50,
-      interval_coverage_90
+      interval_coverage_90,
+      interval_coverage_95
+    ) |>
+    inner_join(summarised_rel,
+      by = c("model", by)
     )
+  return(summarised)
+}
 
-  summarised_scores <- left_join(abs_wis, rel_wis,
-    by = c("model", "location")
-  )
-  return(summarised_scores)
+
+with_horizons <- function(df) {
+  return(df |>
+    mutate(horizon = floor(as.numeric(.data$date -
+      .data$report_date) / 7)))
 }
 
-#' @title Epiweekly Scoring Plot
-#' This function generates a line plot of the weighted interval scores (WIS) for
-#' different models over epiweeks.
-#' @param quantile_scores A scoringutils output table
-#' containing the unsummarised quantile scores.
-#' @param scale A character string specifying the scale
-#' to filter the quantile
-#' scores. Default is "natural".
-#' @return A ggplot object representing the epiweekly scoring plot.
-epiweekly_scoring_plot <- function(quantile_scores, scale = "natural") {
-  epiweekly_score_fig <- quantile_scores |>
-    filter(scale == !!scale) |>
-    mutate(epiweek = epiweek(date), epiyear = epiyear(date)) |>
-    get_pairwise_comparisons(
-      by = c("epiweek", "epiyear"),
-      baseline = "cdc_baseline"
-    ) |>
-    mutate(epidate = forecasttools::epiweek_to_date(
-      epiweek,
-      epiyear,
-      day_of_week = 1
-    )) |>
-    group_by(model, epidate) |>
-    summarise(
-      wis = mean(wis_scaled_relative_skill),
-      .groups = "drop"
-    ) |>
-    as_tibble() |>
-    ggplot(aes(x = epidate, y = wis, color = model)) +
-    geom_line() +
-    geom_point() + # Add points to the line plot
+
+plot_scores_by_date <- function(scores_by_date,
+                                date_column = "date",
+                                score_column = "relative_wis",
+                                model_column = "model",
+                                plot_title = "Scores by model over time",
+                                xlabel = "Date",
+                                ylabel = "Relative WIS") {
+  min_score <- min(scores_by_date[[score_column]])
+  max_score <- max(scores_by_date[[score_column]])
+  max_overall <- max(c(1 / min_score, max_score))
+  sym_ylim <- c(1 / max_overall, max_overall)
```

Comment:
you do this a few times, so there is an opportunity to DRY by creating a `compute_sym_lim` function

Reply:
Yes, I was thinking of dumping it into `forecasttools`